### ISSUE TITLE
Display an info message when mod provider search is loading or complete

### ISSUE THREAD

**Issue by UltraKeelan** [PolyMC/PolyMC#1109]
### Role

 I use PolyMC to manage my mods

### Suggestion

Display an info message when mod provider search is loading or complete

### Benefit

This will make it clear to the user whether PolyMC is waiting for a response from the mod provider, or if it is complete and has no results

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

## Problem

When searching for mods, sometimes certain providers may take a while (upwards of 5 seconds) to return a response. This could happen for a number of reasons, but currently no message is displayed that the client is still waiting for a response. Instead, PolyMC shows an empty results box.

Similarly, when a user searches for a mod and there are no matches, PolyMC again displays an empty results box.

This creates a confusing scenario where users are unsure whether there are no matches for their search, or they're still waiting for the mod provider to return a response for the search query.

## Suggestion

Add the following status messages to the mod search:

* Waiting for response from `mod_provider`.
* Search completed, no results found for `search_term`.
    * Alternatively, a persistent banner message could be shown with something like "`n` results found for search `search_term`.", which may or may not be easier to implement :)

### Future iterations

In the future, it might be nice to add a hint that the results may be affected by filter options, such as:

* No results found for `search_term` for `filter_option`. (i.e. "Strict match (`full_version`)", "Major version match (`major_version`)", "Any version")

And for even more added value, any non-success responses from the server (basically just handling the HTTP responses), such as:

* Search timed out waiting for mod provider (408)
* Mod provider rejected search (401, 403) (this might happen on a blocked VPN IP or something similar)
* Unexpected response from mod provider (other 4xx series)
* Mod provider unavailable (5xx series)

## Benefits

This adds clarity to the user experience, and should hopefully deflect forum posts, issue creation, etc. There's one mod provider in particular that seems to take longer responding to search queries (at least for me), and I never know if I'm still waiting on a response or if my search returned no results.

## Drawbacks

The only real drawback I see here is that, if we add handling for HTTP response codes, the messages need to be phrased very carefully to indicate whether it's a problem with the server or the client, and if the user can/should take any action. If we start displaying `401 Unauthorized` directly to a user with no explanation, someone unfamiliar with HTTP responses might be confused about the source of the issue. In the event that we do add these messages, it might help to also add a link back to a "troubleshooting" page in the wiki.

**Comment by flowln** [comment]
Partially done in #939 by displaying a progress bar when processing the request, and hiding it once the request is done.

**Comment by UltraKeelan** [comment]
> Partially done in #939 by displaying a progress bar when processing the request, and hiding it once the request is done.

@flowln Oh neat, thanks! I did a bit of searching in the issue tracker but I must not have hit the search terms to catch this. There may still be some value to add with error codes and hints about filters, but I'm glad to see that there's already some improvement in the works :)

**Comment by flowln** [comment]
Agreed, the progress bar I added there is very bare! We can definitely do better :)
